src/doc/reference.md

```diff
@@ -1047,11 +1047,8 @@ This is a list of behavior not considered *unsafe* in Rust terms, but that may
 be undesired.
 
 * Deadlocks
-* Reading data from private fields (`std::repr`)
```

Comment:
I think I answered my own question why it's not a good idea to allow [(see forum)](https://internals.rust-lang.org/t/make-invalid-primitive-values-that-are-not-read-valid/2295).